src/models/ConstrainedMetaModel.ts

```diff
@@ -19,6 +19,7 @@ export interface ConstrainedMetaModelOptionsDiscriminator
 export class ConstrainedMetaModelOptions extends MetaModelOptions {
   const?: ConstrainedMetaModelOptionsConst;
   discriminator?: ConstrainedMetaModelOptionsDiscriminator;
+  parentUnions?: ConstrainedUnionModel[];
```

Comment:
Torn between making this a more general `parents` list of all parent models and the specific `parentUnions` required for this feature. This is probably the right call and could be extended if required in future to be `parents`.

Reply:
Didnt even notice it was called parentUnions instead of `parentUnions` 😄 

Would make sense to me yea 👍

Reply:
The other downside of moving this to `parents` would be the need to relax the type and do a bunch of reflection on access. Happy to leave this as it is.

Reply:
`parents` can be handy for other use cases in the future, so I changed it to that. The access for Java's ClassRenderer only needed a check for `model instanceof ConstrainedUnionModel`, so I believe that's perfectly fine.